src/hooks/useMap.ts

```diff
@@ -0,0 +1,146 @@
+import { useEffect, useState } from 'react';
+import { INIT_LOCATION, CENTER_LOCATION } from '@/utils/constants';
+import { RefObject } from 'react';
+import { useDisclosure } from '@chakra-ui/react';
+
+const INIT_ZOOM_LEVEL = 5;
+const MAX_ZOOM_LEVEL = 8;
+
+interface useMapProps {
+  mapRef: RefObject<HTMLElement>;
+}
+
+const useMap = ({ mapRef }: useMapProps) => {
+  const [location, setLocation] = useState(INIT_LOCATION); // 내 디바이스 위치
+  const [map, setMap] = useState<kakao.maps.Map | null>(null);
+  const [zoomLevel, setZoomLevel] = useState(INIT_ZOOM_LEVEL);
+  const [currentCity, setCurrentCity] = useState('서울');
+  const [currentLocation, setCurrentLocation] = useState(INIT_LOCATION); // 지도를 움직이면 변화하는 위치
+
+  useEffect(() => {
+    kakao.maps.load(() => {
+      const success = (pos: GeolocationPosition) => {
+        if (!mapRef.current) return;
+
+        const options = {
+          center: new kakao.maps.LatLng(
+            pos.coords.latitude,
+            pos.coords.longitude
+          ),
+          level: INIT_ZOOM_LEVEL,
+        };
+
+        const kakaoMap = new kakao.maps.Map(mapRef.current, options);
+
+        const currentLocationMarker = new kakao.maps.CustomOverlay({
+          position: new kakao.maps.LatLng(
+            pos.coords.latitude,
+            pos.coords.longitude
+          ),
+          content: `<div class="circle-marker" />`,
+        });
+
+        currentLocationMarker.setMap(kakaoMap);
+        kakaoMap.setMaxLevel(MAX_ZOOM_LEVEL);
+        setMap(kakaoMap);
+      };
+
+      const error = () => {
+        if (!mapRef.current) return;
+
+        const options = {
+          center: new kakao.maps.LatLng(
+            INIT_LOCATION.latitude,
+            INIT_LOCATION.longitude
+          ),
+          level: INIT_ZOOM_LEVEL,
+        };
+
+        const kakaoMap = new kakao.maps.Map(mapRef.current, options);
+
+        const currentLocationMarker = new kakao.maps.CustomOverlay({
+          position: new kakao.maps.LatLng(
+            INIT_LOCATION.latitude,
+            INIT_LOCATION.longitude
+          ),
+          content: `<div class="circle-marker" />`,
+        });
+
+        currentLocationMarker.setMap(kakaoMap);
+        kakaoMap.setMaxLevel(MAX_ZOOM_LEVEL);
+        setMap(kakaoMap);
+      };
+      navigator.geolocation.getCurrentPosition(success, error);
+    });
+  }, []);
+
+  useEffect(() => {
+    if (!map) return;
+
+    kakao.maps.event.addListener(map, 'zoom_changed', () => {
+      const zoomLevel = map.getLevel();
+      setZoomLevel(zoomLevel);
+    });
+
+    const geocoder = new kakao.maps.services.Geocoder();
+
+    kakao.maps.event.addListener(map, 'dragend', function () {
+      const coords = map.getCenter();
+      const [latitude, longitude] = [coords.getLat(), coords.getLng()];
+      setCurrentLocation({ latitude, longitude });
```

Comment:
```suggestion
    kakao.maps.event.addListener(map, 'dragend', () => {
      const coords = map.getCenter();
      const [latitude, longitude] = [coords.getLat(), coords.getLng()];
      setCurrentLocation({ latitude, longitude });
```
화살표 함수로 통일하면 좋을 것 ㄱ타아요